### PR TITLE
Chore: Migrate ForwardLivechatView to Typescript

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

Migrate ForwardLivechatView to Typescript, besides I created a file with definitions for livechat's departments.

## How to test or reproduce

- For the test, you will need to be at least a livechat agent
- Then, in an omnichannel's room, click on the Title in header
- Now you can click on the `Forward` button
- The layout and functions still working properly

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/ForwardLivechatView.tsx [hunk 1]
```diff
@@ -85,11 +122,14 @@ const ForwardLivechatView = ({ forwardRoom, navigation, route, theme }) => {
...
 	useEffect(() => {
+		navigation.setOptions({
+			title: I18n.t('Forward_Chat')
+		});
 		getRoom();
 	}, []);
 
 	useEffect(() => {
-		if (room) {
+		if (!isEmpty(room)) {
```
Comment (reinaldonetof) [inline app/views/ForwardLivechatView.tsx:132]: I changed this line to working properly with the `useState({})`

File: app/views/ForwardLivechatView.tsx [hunk 2]
```diff
@@ -18,20 +22,52 @@ const styles = StyleSheet.create({
...
+	const [room, setRoom] = useState<IRoom>({});
 
 	const rid = route.params?.rid;
 
 	const getDepartments = async () => {
 		try {
 			const result = await RocketChat.getDepartments();
 			if (result.success) {
-				setDepartments(result.departments.map(department => ({ label: department.name, value: department._id })));
+				const deps = result.departments as ILivechatDepartment[];
```
Comment (reinaldonetof) [inline app/views/ForwardLivechatView.tsx:69]: What do you think about this line and the line 86?

Comment (AlexAlexandre) [reply app/views/ForwardLivechatView.tsx:69]: Why you choose do it like you did as like this?
```const deps: ILivechatDepartment[] = result.departments;```  

Change the line 86 for the same pattern, please :)

File: app/views/ForwardLivechatView.tsx (hunk 1 above)
Comment (AlexAlexandre) [reply app/views/ForwardLivechatView.tsx:132]: revert this change please :)

**Review by AlexAlexandre — CHANGES_REQUESTED** [review]

**Review by AlexAlexandre — APPROVED** [review]
LGTM :)

**Pull request merged**